src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -121,15 +122,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   private Entity createLineEntity(Node node, long lectureId) {
     Element element = (Element) node;
     String lineContent = node.getTextContent();
-    String lineStart = element.getAttribute(START_ATTRIBUTE);
-    String lineDuration = element.getAttribute(DURATION_ATTRIBUTE);
-
+    Float lineStart = Float.parseFloat(element.getAttribute(START_ATTRIBUTE));
+    // java.time.duration is not a supported type for data store. Thus, the line duration
+    // is temporarily stored as it was received when it was parsed from the transcript XML.
+    // TODO: Find a better way to store duration or remove it entirely.
+    Float lineDuration = Float.parseFloat(element.getAttribute(DURATION_ATTRIBUTE));
+    Float lineEnd = lineStart.floatValue() + lineDuration;
     Entity lineEntity = new Entity(TranscriptLine.ENTITY_KIND);
+
     lineEntity.setProperty(
         TranscriptLine.PROP_LECTURE, KeyFactory.createKey(PARAM_LECTURE, lectureId));
     lineEntity.setProperty(TranscriptLine.PROP_CONTENT, lineContent);
-    lineEntity.setProperty(TranscriptLine.PROP_START, lineStart);
+    lineEntity.setProperty(
+        TranscriptLine.PROP_START, new Date(lineStart.longValue() * MILLISECOND_CONVERTER));
     lineEntity.setProperty(TranscriptLine.PROP_DURATION, lineDuration);
+    lineEntity.setProperty(
+        TranscriptLine.PROP_END, new Date(lineEnd.longValue() * MILLISECOND_CONVERTER));
```

Comment:
I believe there are recommended functions to convert between seconds and ms.  There can be issues doing it manually with multiplication (if there are overflows or something... probs not a worry here, but good practice)